impl/src/main/java/jakarta/faces/context/FacesContext.java

```diff
@@ -62,8 +63,8 @@ public abstract class FacesContext {
     private boolean processingEvents = true;
     private boolean isCreatedFromValidFactory = true;
 
-    private static ConcurrentHashMap threadInitContext = new ConcurrentHashMap(2);
-    private static ConcurrentHashMap initContextServletContext = new ConcurrentHashMap(2);
+    private static final ConcurrentHashMap<Thread,FacesContext> threadInitContext = new ConcurrentHashMap<>(2);
+    private static final ConcurrentHashMap<Thread,InitFacesContext> initContextServletContext = new ConcurrentHashMap<>(2);
```

Comment:
FYI: this caused a major problem during 4.1.2 release.

Please never import impl specific classes into API classes :)